packages/rum-core/src/domain/rumEventsCollection/error/errorCollection.ts

```diff
@@ -43,6 +44,12 @@ export function doStartErrorCollection(lifeCycle: LifeCycle, observable: Observa
   }
 }
 
+export function createRumErrorFilter(lifeCycle: LifeCycle, configuration: Configuration) {
+  return createErrorFilter(configuration, (error) => {
+    lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, processError(error))
+  })
+}
```

Comment:
From a design perspective, I think it could be more interesting to:
- create the error filter in assembly, since it is not really related to error collection
- introduce a RAW_ERROR_COLLECTED lifeCycle event to allow to generate errors from other places

Reply:
hum, lifeCycle events seems quite consistent with what we are doing, so not sure why we should do something different here or what is the issue with introducing a new lifeCycle event.
I am also not a big fan of introducing an error collection dependency in assembly.

Reply:
> what is the issue with introducing a new lifeCycle event

LifeCycle events are global, so any new LifeCycle event is complexifying the overall design of RUM. Having only a few important events makes things easier to explain and could be represented as a graph if we want to. We chose to have a few more events to decouple modules allowing to write tests without having to mock them. Here, we don't need to listen to a potential `RAW_ERROR_COLLECTED` event in tests, so IMO introducing a new lifecycle event for this is not necessary.

>  I am also not a big fan of introducing an error collection dependency in assembly.

Here, the only requirement is to transform the 'raw error' into an 'raw rum error event'. It could be moved to another shared module.

Anyway, it's not a big deal, I'll add the RAW_ERROR_COLLECTED event.

Reply:
it could worth to talk it through to be aligned on that though